quinn-udp/src/unix.rs

```diff
@@ -66,6 +66,100 @@ impl Default for UdpSocketState {
     }
 }
 
+/// Fallback implementation of `sendmmsg` using `sendmsg`
+/// for systems which do not support `sendmmsg`
+/// such as Linux <3.0.
+#[cfg(not(any(target_os = "macos", target_os = "ios")))]
+unsafe fn sendmmsg_fallback(
+    sockfd: libc::c_int,
+    msgvec: *mut libc::mmsghdr,
+    vlen: libc::c_uint,
+) -> libc::c_int {
+    let flags = 0;
+    if vlen == 0 {
+        return 0;
+    }
+
+    let n = libc::sendmsg(sockfd, &(*msgvec).msg_hdr, flags);
+    if n == -1 {
+        -1
+    } else {
+        (*msgvec).msg_len = n as libc::c_uint;
+        1
+    }
+}
+
+#[cfg(not(any(target_os = "macos", target_os = "ios")))]
+unsafe fn sendmmsg_with_fallback(
+    sockfd: libc::c_int,
+    msgvec: *mut libc::mmsghdr,
+    vlen: libc::c_uint,
+) -> libc::c_int {
+    let flags = 0;
+    let ret = libc::syscall(libc::SYS_sendmmsg, sockfd, msgvec, vlen, flags) as libc::c_int;
```

Comment:
Out of curiosity, is the difference between `libc::syscall(libc::SYS_recvmmsg, ..)` and `libc::recvmmsg(..)` material for catching an `ENOSYS` error?

Reply:
Ah, makes sense. Maybe add a short comment to explain the rationale behind using `syscall()` here?